### PR TITLE
[PoC] add support for passing rng to transforms

### LINKED ISSUE #7027: Transforms V2 proposal: Enabling reproducible workflows with local generators
### 🚀 The feature

(This was originally [pitched in this long feedback thread](https://github.com/pytorch/vision/issues/6753#issuecomment-1304296899). It was recommended that I open a separate issue).

Enable the new transforms API to support the use of local generators to control RNG via the modified API:

```python
from torch import Generator, Tensor, default_generator
import torch.nn as nn

class Transform(nn.Module):
    def _get_params(self, flat_inputs: List[Any], *, generator: Generator) -> Dict[str, Any]:
        ...

    def forward(self, *inputs: Any, generator: Generator = default_generator) -> Any:
        # the only modification
        params = self._get_params(flat_inputs, generator=generator)
```

Thus transforms that implement `_get_params` would replace calls like

```python
# e.g. replace calls like
angle = float(torch.empty(1).uniform_(0.0, 180.).item())
```

with 

```python
# specifying the device is, unfortunately, necessary: https://github.com/pytorch/pytorch/issues/79018
angle = float(torch.empty(1, device=generator.device).uniform_(0.0, 180., generator=generator).item())
```

### Motivation, pitch

In recent years, [NumPy has completely revised their PRNG API](https://numpy.org/devdocs/reference/random/index.html) to avoid global random state ([here](https://albertcthomas.github.io/good-practices-random-number-generators/) is a great post on good practices with NumPy's generators). [JAX avoids mutable RNG objects altogether](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#random-numbers). PyTorch provides `torch.Generator` to users to to make randomness local and "non-spooky", but many libraries prevent users from utilizing this capability.

I am proposing that `Transform` enable users to optionally pass in a `Generator` to the forward pass so that torchvision transform pipelines can be made to be isolated from global entropy and thus support more reproducible workflows. This reproducibility is especially useful in the context of performing testing & evaluation – the specific sequence of data transformations performed should be able to be isolated from whether or not a model is using dropout in its forward pass.

### Alternatives

_No response_

### Additional context

@pmeier  already provided (positive) feedback on this proposal [here](https://github.com/pytorch/vision/issues/6753#issuecomment-1304343868)

### PR DESCRIPTION
**Pull request by pmeier**
Closes #7027.

### CODE REVIEW

**Comment by pmeier** [comment]
As discussed in #7027, passing a generator to `forward` is impossible when using the transforms the regular way through a dataset. Meaning, we need to have it as an instance attribute.

If we go that route, there is little to salvage here.
